From: Hannes Reinecke <hare@suse.de>
Date: Mon, 29 Sep 2014 15:22:37 +0200
Subject: [PATCH] scsi: clear 'host_scribble' upon successful abort
References: bnc#894863
Patch-Mainline: Depends on patches.fixes/scsi-Warn-on-invalid-command-completion.patch

When an asynchronous abort succeeds we should be clearing
'host_scribble' to not trip over a WARN_ON() during
scsi_put_command().

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/scsi_error.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/scsi/scsi_error.c b/drivers/scsi/scsi_error.c
index 84b8dc2..b1055b5 100644
--- a/drivers/scsi/scsi_error.c
+++ b/drivers/scsi/scsi_error.c
@@ -132,6 +132,7 @@ scmd_eh_abort_handler(struct work_struct *work)
 		rtn = scsi_try_to_abort_cmd(sdev->host->hostt, scmd);
 		if (rtn == SUCCESS) {
 			scmd->result |= DID_TIME_OUT << 16;
+			scmd->host_scribble = NULL;
 			if (scsi_host_eh_past_deadline(sdev->host)) {
 				SCSI_LOG_ERROR_RECOVERY(3,
 					scmd_printk(KERN_INFO, scmd,
-- 
1.8.5.2

